### PR TITLE
Refactor all elements to use Twig

### LINKED ISSUE #88: Cache invalidation on associated elements
When a block is associated with a view and the block gets updated, it doesn't invalidate the view, which means the view is not invalidated until the TTL kicks in

### LINKED ISSUE #89: Carousel elements are not rendered by position
I have this carousel:

image

but in the frontend:

image

it still starts off by showing slide 1

### PR DESCRIPTION
**Pull request by loevgaard**
Fixes #88 
Resolves #89

- [x] ~~Provide debug info when in dev mode (as it was the old way)~~ Instead we just log an error to the logger. The debug info isn't worth much in production anyway
- [x] Add elements to element stack (this was previously done in renderers)
- [x] Extend `TimestampableInterface` on `ElementInterface`. This will be used in the element loaders

### CODE REVIEW

File: src/Doctrine/ORM/ViewRepository.php [hunk 1]
```diff
@@ -4,25 +4,20 @@
...
-        $view = $this->findOneBy([
-            'code' => $code,
-        ]);
+        $obj = parent::findOneByCode($code);
 
-        Assert::nullOrIsInstanceOf($view, ViewInterface::class);
+        Assert::nullOrIsInstanceOf($obj, ViewInterface::class);
 
-        return $view;
+        return $obj;
```
Comment (Roshyo) [inline src/Doctrine/ORM/ViewRepository.php:20]: I'd prefer to avoid shortcut, but ok :D

File: src/Generator/Twig/CarouselTwigGenerator.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        $items = [];
+
+        /** @var CarouselBlockInterface $carouselBlock */
+        foreach ($element->getCarouselBlocks() as $carouselBlock) {
+            $block = $carouselBlock->getBlock();
+            if (null === $block) {
+                continue;
+            }
+
+            $items[$carouselBlock->getPosition()] = (string) $block->getCode();
```
Comment (Roshyo) [inline src/Generator/Twig/CarouselTwigGenerator.php:34]: Can't it happen that 2 blocks have the same position ?

File: src/Generator/Twig/ViewTwigGenerator.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        );
+
+        /** @var array<string, array<int, string>> $blocks */
+        $blocks = [];
+
+        foreach ($element->getViewBlocks() as $viewBlock) {
+            $block = $viewBlock->getBlock();
+            Assert::notNull($block);
+
+            $blocks[(string) $viewBlock->getSection()][$viewBlock->getPosition()] = (string) $block->getCode(); // todo notice that if somehow two blocks end up at the same position, this will effectively only show one of them, so this needs to be checked when saving a view or fixed some other way
```
Comment (Roshyo) [inline src/Generator/Twig/ViewTwigGenerator.php:39]: What about using gedmo sortable?

File: src/Twig/Loader/ElementLoader.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+namespace Setono\SyliusCMSPlugin\Twig\Loader;
+
+use InvalidArgumentException;
+use Setono\SyliusCMSPlugin\Twig\Loader\ElementLoader\ElementLoaderInterface;
+use Setono\SyliusCMSPlugin\Twig\Loader\ElementLoader\LogicalTemplateName;
+use Twig\Error\LoaderError;
+use Twig\Loader\LoaderInterface;
+use Twig\Source;
+
+final class ElementLoader implements LoaderInterface
```
Comment (loevgaard) [inline src/Twig/Loader/ElementLoader.php]: Should probably be renamed to `CompositeElementLoader` or something to not add confusion about the plugin-native loaders in the `Setono\SyliusCMSPlugin\Twig\Loader\ElementLoader` namespace

File: src/Resources/views/carousel.html.twig [hunk 5]
```diff
@@ -1 +1,47 @@
...
+                slidesToScroll: {{ configuration.slidesToScroll }},
+                prevArrow: $('.{{ identifier }} button.left'),
+                nextArrow: $('.{{ identifier }} button.right'),
+                appendArrows: false,
+                autoplay: {{ configuration.autoplay ? 'true' : 'false' }},
+                autoplaySpeed: {{ configuration.autoplaySpeed }},
+            });
+        });
+    </script>
+{% endblock %}
```
Comment (Roshyo) [inline src/Resources/views/carousel.html.twig:47]: Not sure about jquery loading here :thinking:

File: src/Doctrine/ORM/ViewRepository.php (hunk 1 above)
Comment (loevgaard) [reply src/Doctrine/ORM/ViewRepository.php:20]: I like this approach. It helps when you copy paste 😂

File: src/Generator/Twig/ViewTwigGenerator.php (hunk 3 above)
Comment (loevgaard) [reply src/Generator/Twig/ViewTwigGenerator.php:39]: Sounds like a good idea. I have never really used it and I think I can recall a couple of issues in former projects where it didn't work as expected. Do you have good experience with it?

File: src/Resources/views/carousel.html.twig (hunk 5 above)
Comment (loevgaard) [reply src/Resources/views/carousel.html.twig:47]: Wasn't it you creating the carousels? :D But anyways, this is just for people to override and to show in the plugin application. I guess it's okay?

**Review by Roshyo — APPROVED** [review]

File: src/Generator/Twig/ViewTwigGenerator.php (hunk 3 above)
Comment (Roshyo) [reply src/Generator/Twig/ViewTwigGenerator.php:39]: Well, I just used it a couple of times, and it works perfectly. You can even set priority groups, and the priority will be done according to another column as well

**Comment by loevgaard** [comment]
I have added issue for your questions, @Roshyo so they are not forgotten :)

**Pull request merged**